perun-core/src/main/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntry.java

```diff
@@ -139,4 +145,27 @@ public ConsentHub updateConsentHub(PerunSession sess, ConsentHub consentHub) thr
 
 		return getConsentsManagerBl().updateConsentHub(sess, consentHub);
 	}
+	@Override
+	public Consent changeConsentStatus(PerunSession sess, int consentId, ConsentStatus status, int userId) throws ConsentNotExistsException, PrivilegeException, UserNotExistsException, InvalidConsentStatus {
+		Utils.notNull(sess, "sess");
+
+		Consent consent = getPerunBl().getConsentsManagerBl().getConsentById(sess, consentId);
+		User user = getPerunBl().getUsersManager().getUserById(sess, userId);
+
+		if (status != ConsentStatus.GRANTED && status != ConsentStatus.REVOKED) {
```

Comment:
These checks and the logic of setting the status should probably be in BL layer. (not sure, check with @Johaney-s  or @cuadradek )

Reply:
Yeah, I agree